src/Components/AdvancedSearch/AdvancedSearchModal.tsx

```diff
@@ -43,11 +42,35 @@ const AdvancedSearchModal: React.FC<IAdvancedSearchProps> = (props) => {
     });
     const { t } = useTranslation();
     const titleId = useId('advanced-search-modal-title');
+    const filteredTwins = useRef<IADTTwin[]>([]);
     const additionalProperties = useRef(new Set<string>());
-    // TODO: Make this actually call query
+    const searchForTwinAdapterData = useAdapter({
+        adapterMethod: (params: AdapterMethodParamsForAdvancedSearchADTwins) =>
+            adapter.advancedSearchADTTwins(params),
+        refetchDependencies: [],
+        isAdapterCalledOnMount: false
+    });
+
     const executeQuery = (query: string) => {
-        alert(query);
+        searchForTwinAdapterData.callAdapter({
+            query: query,
+            // TODO: Define onScroll method on list to change continuation token
+            continuationToken: null
```

Comment:
we need to keep track of the continuationToken and pass it here when you do the pagination, right?

Reply:
I have asked him to remove this from the component level and keep it contained inside the adapter. At this point we don't plan to do pagination on the control level so to get all the data we have to do taht recursion internally. Though it might be in the next PR.

Reply:
removed